requests_opentracing/__init__.py

```diff
@@ -0,0 +1,105 @@
+from functools import wraps
+
+import opentracing
+import requests
+
+
+g_tracer = None
+g_trace_all = True
+g_start_span_cb = None
+
+
+def init_tracing(tracer, trace_all=True, start_span_cb=None):
+    '''
+    Set our tracer for requests instrumentation. Tracer objects
+    from the OpenTracing django/flask/pyramid libraries can be
+    passed as well.
+
+    :param tracer: the tracer object.
+    :param trace_all: If True, all requests are automatically traced.
+        Else, explicit tracing on Session objects is required.
+    :param start_span_cb: The callback function invoked when
+        a new Span is created, having the Span and the request object
+        as arguments. Can be used to set the operation name or set
+        custom tags.
+    '''
+    global g_tracer, g_trace_all, g_start_span_cb
+
+    if hasattr(tracer, '_tracer'):
+        tracer = tracer._tracer
+
+    if tracer is None:
+        raise ValueError('tracer')
+
+    if start_span_cb is not None and not callable(start_span_cb):
+        raise ValueError('start_span_cb')
+
+    g_tracer = tracer
+    g_trace_all = trace_all
+    g_start_span_cb = start_span_cb
+
+    if trace_all:
+        _patch_session_send()
+
+
+def trace_session(session):
+    '''
+    Marks a Session to be traced. All requests executed
+    through this object will be traced.
+    '''
+    if session is None:
+        raise ValueError('session')
+
+    send_method = session.send
+
+    @wraps(send_method)
+    def tracing_send(request, **kwargs):
+        return _call_send(None, send_method, request, **kwargs)
+
+    session.send = tracing_send
+
+
+def _patch_session_send():
+    send_method = requests.Session.send
+
+    @wraps(send_method)
+    def tracing_send(self, request, **kwargs):
+        return _call_send(self, send_method, request, **kwargs)
+
+    requests.Session.send = tracing_send
+
+
+def _call_send(self, send, request, **kwargs):
+    span = g_tracer.start_span(request.method)
```

Comment:
Actually, upon checking the `opentracing` and `basictracer` packages for Python, I realized the context manager is by default 'logging' some error tags:
```
        if exc_type:
            self.log_kv({
                'python.exception.type': exc_type,
                'python.exception.val': exc_val,
                'python.exception.tb': exc_tb,
                })
```

Since `error` and `error.object` are the documented ones, should we create a PR there for having such too? And then we could simply use a manual `try... finally` block *only* if additional tags (or scenarios) are required ;) Let me know.